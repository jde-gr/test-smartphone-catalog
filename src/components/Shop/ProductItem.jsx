import { Link } from 'react-router-dom';

import Card from '../UI/Card';
import {
  Brand,
  Content,
  Image,
  ImgContainer,
  LinkContainer,
  Model,
  Price,
} from './styles/ProductItem.styles';

const ProductItem = (props) => {
  const { id, brand, model, price, imgUrl } = props;
  return (
    <Card>
      <LinkContainer>
        <Link
          to={`/device/:${id}`}
          state={{ id: id, device: `${brand} ${model}` }}
          data-testid={id}
        >
          <ImgContainer>
            <Image src={imgUrl} alt={`Dispositivo ${brand} ${model}`} />
          </ImgContainer>
          <Content>
            <Brand>Marca: {brand}</Brand>
            <Model>Modelo: {model}</Model>
            <Price>Precio: {price}â‚¬</Price>
          </Content>
        </Link>
      </LinkContainer>
    </Card>
  );
};

export default ProductItem;
