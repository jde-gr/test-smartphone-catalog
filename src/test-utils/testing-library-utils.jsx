import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Provider>
    </BrowserRouter>
  );
};
const renderWithContext = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithContext as render };
