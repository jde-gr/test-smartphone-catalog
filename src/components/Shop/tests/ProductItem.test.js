import { render } from '../../../test-utils/testing-library-utils';
import ProductItem from '../ProductItem';

test('renders correctly', () => {
  const { container } = render(<ProductItem />);
  expect(container).toMatchSnapshot();
});
