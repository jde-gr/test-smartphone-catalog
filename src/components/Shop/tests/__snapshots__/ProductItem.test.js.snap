// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    style="width: inherit;"
  >
    <div
      class="css-1xb5uqo"
    >
      <div
        class="css-1b8ia93"
      >
        <a
          href="/device/:undefined"
        >
          <div
            class="css-1r4ryte"
          >
            <img
              alt="Dispositivo undefined undefined"
              class="css-0"
            />
          </div>
          <div
            class="css-ihzpm"
          >
            <h4
              class="css-1iv4gsg"
            >
              Marca: 
            </h4>
            <h6
              class="css-bhh5v6"
            >
              Modelo: 
            </h6>
            <p
              class="css-1kkf7xj"
            >
              Precio: 
              â‚¬
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
